src/server/controllers/AuthController.js

```diff
@@ -23,36 +27,80 @@ class AuthController {
       try {
         validateSignupPayload(req.body);
       } catch (error) {
-        error.code = 422;
         return next(error);
       }
       const { username, email, password } = req.body;
       return models.User.create({ username, email, password })
-        .then((user) => {
-          req.user = extractUserInfo(user);
-          return next();
-        })
+        .then(user =>
+          generateToken(user, rsaKey)
+            .then((token) => {
+              const userInfo = extractUserInfo(user);
+              return res.status(200).json({
+                user: {
+                  id: userInfo.id,
+                  username: userInfo.username,
+                  email: userInfo.email
+                },
+                token,
+                message: 'User created successfully'
+              });
+            }))
         .catch(error => next(error));
     };
   }
 
   /**
-   * Authenticate a user with username and password
+   * Authenticates user token
+   * @method
+   * @memberof AuthController
+   * @static
+   * @return {function} Express middleware function that calls the
+   * next method after authentication has been completed
+   */
+  static isAuthenticated() {
```

Comment:
I'm guessing there's a mixup here. This controller is not needed for signup and signin functionalities. This should come in subsequently when user is making a request to access a resource.
And there's the `getClientToken` and `authorizeUser` controllers for that already.

Reply:
didn't realize that has been added. Will fix it.